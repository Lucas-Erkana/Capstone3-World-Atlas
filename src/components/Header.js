import { Link } from 'react-router-dom';

const Header = () => (
  <header>
    <Link className="link" to="/">
      {' '}
      <i className="fa-solid fa-house" />
    </Link>
    <h2 className="header-tag">The World At Your Feet</h2>
    <div className="header-controls">
      <i className="fa-solid fa-microphone fa-2x" />
      <i className="fa-solid fa-gear fa-2x" />
    </div>
  </header>
);

export default Header;
