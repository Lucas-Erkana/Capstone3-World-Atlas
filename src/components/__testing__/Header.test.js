import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import store from '../../redux/configureStore';
import Header from '../Header';

describe('Testing the Header component', () => {
  it('The Header component should match the snapshot: ', () => {
    const picker = render(
      <BrowserRouter>
        <Provider store={store}>
          <Header />
        </Provider>
      </BrowserRouter>,
    );
    expect(picker).toMatchSnapshot();
  });
});
