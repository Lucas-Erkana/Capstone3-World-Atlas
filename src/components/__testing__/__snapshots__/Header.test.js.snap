// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Header component The Header component should match the snapshot:  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <a
          class="link"
          href="/"
        >
           
          <i
            class="fa-solid fa-house"
          />
        </a>
        <h2
          class="header-tag"
        >
          Explore The World
        </h2>
        <div
          class="header-controls"
        >
          <i
            class="fa-solid fa-microphone fa-2x"
          />
          <i
            class="fa-solid fa-gear fa-2x"
          />
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header>
      <a
        class="link"
        href="/"
      >
         
        <i
          class="fa-solid fa-house"
        />
      </a>
      <h2
        class="header-tag"
      >
        Explore The World
      </h2>
      <div
        class="header-controls"
      >
        <i
          class="fa-solid fa-microphone fa-2x"
        />
        <i
          class="fa-solid fa-gear fa-2x"
        />
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
